File: Conversation_Outputs/Convergence_Tests/QuixBugs_Java/java_programs/KHEAPSORT.java
package java_programs;
import java.util.*;

public class KHEAPSORT {

    public static ArrayList<Integer> kheapsort(ArrayList<Integer> arr, int k) {
        PriorityQueue<Integer> heap = new PriorityQueue<Integer>();
        
        // Initialize the heap with the first k elements
        for (Integer v : arr.subList(0, k)) {
            heap.add(v);
        }

        ArrayList<Integer> output = new ArrayList<Integer>();

        // Start the loop from the kth element, as the first k elements are already in the heap
        for (int i = k; i < arr.size(); i++) {
            heap.add(arr.get(i)); // Add the current element to the heap
            Integer popped = heap.poll(); // Poll the smallest element
            output.add(popped); // Add it to the output list
        }

        // Poll remaining elements from the heap
        while (!heap.isEmpty()) {
            output.add(heap.poll());
        }

        return output;
    }
}

File: Conversation_Outputs/Guided_Convergence_Code/Python/kheapsort.py
def kheapsort(arr, k):
    import heapq

    # Initialize the heap with the first k elements
    heap = arr[:k]
    heapq.heapify(heap)

    # Start iterating from the k-th element since the first k elements are already in the heap
    for x in arr[k:]:
        yield heapq.heappushpop(heap, x)

    # Yield the remaining elements in the heap
    while heap:
        yield heapq.heappop(heap)